### PR TITLE
Improve: return well formatted errors where API allows

### PR DESCRIPTION
**Pull request by vkhoroz**
Most of our APIs return the well-formatted JSON error for 4xx status codes. Print these errors in a user-friendly output.

Signed-off-by: Volodymyr Khoroz <volodymyr.khoroz@foundries.io>

### CODE REVIEW

**Comment by doanac** [comment]
this is a welcome change I've wanted to introduce forever. you mind sharing a couple examples of the new output in the commit message?

File: client/foundries.go [hunk 1]
```diff
@@ -542,10 +542,58 @@ func readResponse(res *http.Response, log logrus.FieldLogger) (*[]byte, error) {
...
+			}
+			dictErrors struct {
+				Msg     string            `json:"msg,omitempty"`
+				Message string            `json:"message,omitempty"`
+				Errors  map[string]string `json:"errors,omitempty"`
+			}
+		)
+
+		if merr := json.Unmarshal(body, &listErrors); merr == nil {
+			if listErrors.Msg != "" {
```
Comment (mike-sul) [inline client/foundries.go:562]: Just really not important comment. The first part for these two types of Errors is the same so maybe the code can be unified.

**Review by mike-sul — APPROVED** [review]

File: client/foundries.go (hunk 1 above)
Comment (mike-sul) [reply client/foundries.go:562]: My first thought was about doing it in a functional way :).

File: client/foundries.go [hunk 2]
```diff
@@ -542,10 +542,58 @@ func readResponse(res *http.Response, log logrus.FieldLogger) (*[]byte, error) {
...
+				msg += "\n=" + dictErrors.Msg
+			} else if dictErrors.Message != "" {
+				msg += "\n=" + dictErrors.Message
+			} else {
+				useGenericError = true
+			}
+			if !useGenericError && dictErrors.Errors != nil {
+				for field, emsg := range dictErrors.Errors {
+					msg += fmt.Sprintf("\n * %s: %s", field, emsg)
+				}
```
Comment (doanac) [inline client/foundries.go:585]: It's kind of nitpicking, but the `=<message` felt a little weird to me. e.g.:
```
ERROR: HTTP error during PATCH 'https://api.foundries.io/ota/devices/andy-quitt/config/?factory=andy-corp': 400 BAD REQUEST
=Given tag 'foooo' is not present in targets.json
```

I wonder if we should use the asterisk approach for all of this and make the message be:
```
ERROR: HTTP error during PATCH 'https://api.foundries.io/ota/devices/andy-quitt/config/?factory=andy-corp': 400 BAD REQUEST
* Given tag 'foooo' is not present in targets.json
```

Code seems fine though.

Comment (vkhoroz) [reply client/foundries.go:585]: We use `= ` in fioctl in other locations like this; so I preserve the same style here.
The ` *` is actually a _new way_ which is only used here, and only to represent a list of items... message is printed the same as elsewhere.

That said, there is a small problem - I need to ad the _space after =_ (it was missing in the initial commit too, but is present in other locations).

Comment (vkhoroz) [reply client/foundries.go:585]: changed in force-push

**Pull request merged**